packit_service/service/api/usage.py

```diff
@@ -817,3 +820,84 @@ def get(self):
         delta_hours = int(escape(request.args.get("hours", "0")))
         delta_days = int(escape(request.args.get("days", "0")))
         return _get_usage_interval_data(hours=delta_hours, days=delta_days, count=count)
+
+
+def packit_prs_merged(project: GitProjectModel, downstream_projects: bool = True):
+    """Get the list of project's merged PR belonging to Packit.
+    By default it looks only on downstream git projects.
+    """
+    from packit_service.config import ServiceConfig
+    from ogr.abstract import PRStatus
+
+    downstream_project_url = f"https://{project.instance_url}"
+    if (
+        downstream_projects
+        and downstream_project_url != "https://src.fedoraproject.org"
+    ):
+        return []
+
+    pagure_service = [
+        service
+        for service in ServiceConfig.get_service_config().services
+        if service.instance_url == downstream_project_url
+    ][0]
+    ogr_project = pagure_service.get_project(
+        namespace=project.namespace,
+        repo=project.repo_name,
+    )
+    prs = ogr_project.get_pr_list(status=PRStatus.merged)
+    prs_from_packit = [pr for pr in prs if pr.author in ("packit", "packit-stg")]
+    return prs_from_packit
+
+
+@usage_ns.route("/onboarded-projects")
+class Onboarded(Resource):
+    @ttl_cache(maxsize=_CACHE_MAXSIZE, ttl=timedelta(days=1).seconds)
+    @usage_ns.response(
+        HTTPStatus.OK,
+        "Onboarded projects for which exist a Bodhi update or a Koji build or a Packit merged PR.",
+    )
+    def get(self):
+        """
+        Returns a list of onboarded projects for which exist at least a
+        Bodhi update, a downstream Koji build or a merged Packit PR.
+
+        Examples:
+        /api/usage/onboarded-projects
+        """
+        known_onboarded_projects = GitProjectModel.get_known_onboarded_projects()
+
+        bodhi_updates = BodhiUpdateTargetModel.get_all_projects()
+        koji_builds = KojiBuildTargetModel.get_all_projects()
+        onboarded_projects = bodhi_updates.union(koji_builds)
+
+        sync_release_pr = SyncReleaseTargetModel.get_all_projects()
+        downstream_sync_projects = {
+            project
+            for project in sync_release_pr
+            if project.project_url.startswith("https://src.fedoraproject.org")
+        }
```

Comment:
I know this PR is still a draft, so just one important note:

if I get it correctly, the `get_all_projects` gets the projects related to the trigger (=project_event), which in case of sync releases are mostly upstream projects (as the trigger is mostly an upstream release, only with retriggers it is not), so I think with this we will get only the downstream projects where there was some retrigger via dist-git PR comment

I would say if we worked only with our DB, we would need to check `downstream_pr_url` and get the corresponding dist-git project from there

Reply:
I see, I think you are right, Thanks!